geode-core/src/main/java/org/apache/geode/distributed/internal/InternalDistributedSystem.java

```diff
@@ -383,15 +387,31 @@ public static InternalDistributedSystem newInstance(Properties config,
    */
   public static InternalDistributedSystem newInstanceForTesting(DistributionManager dm,
       Properties nonDefault) {
-    InternalDistributedSystem sys = new InternalDistributedSystem(nonDefault);
-    sys.config = new RuntimeDistributionConfigImpl(sys);
-    sys.dm = dm;
-    sys.isConnected = true;
-    return sys;
+    return newInstanceForTesting(dm, nonDefault, defaultStatisticsManagerFactory(),
+        new CompositeMetricsCollector());
   }
 
-  public static InternalDistributedSystem newInstanceForTesting(StatisticsManagerFactory factory) {
-    return new InternalDistributedSystem(new Properties(), factory);
+  static InternalDistributedSystem newInstanceForTesting(
+      DistributionManager dm, Properties properties,
+      StatisticsManagerFactory statisticsManagerFactory, MetricsCollector meterManager) {
+    InternalDistributedSystem internalDistributedSystem =
+        new InternalDistributedSystem(properties, statisticsManagerFactory, meterManager);
+    internalDistributedSystem.config = new RuntimeDistributionConfigImpl(internalDistributedSystem);
+    internalDistributedSystem.dm = dm;
+    internalDistributedSystem.isConnected = true;
+    internalDistributedSystem.configureMeterRegistry();
+    return internalDistributedSystem;
+  }
+
+  private void configureMeterRegistry() {
```

Comment:
It looks like this non-static is in the middle of the "static methods" section in this class. Maybe it should be moved lower?

Reply:
I'm thinking of moving the tag-adding stuff into the `CompositeMetricsCollector`. If we do that, what's left in `configureMeterRegistry()` might be small enough to inline. Otherwise, I'll move it out of the static methods section. And rename it `configureMetricsCollector()`.